src/components/clusterConfiguration/utils.ts

```diff
@@ -61,40 +57,19 @@ export const getSubnetFromMachineNetworkCidr = (machineNetworkCidr?: string) =>
   return getHumanizedSubnet(subnet);
 };
 
-const getDefaultNetworkSettings = (clusterNetworkCidr: string): ClusterNetworkDefaultSettings => {
-  const isIPv4 = Address4.isValid(clusterNetworkCidr);
-  return {
-    clusterNetworkCidr: isIPv4
-      ? CLUSTER_DEFAULT_NETWORK_SETTINGS_IPV4.clusterNetworkCidr
-      : CLUSTER_DEFAULT_NETWORK_SETTINGS_IPV6.clusterNetworkCidr,
-    serviceNetworkCidr: isIPv4
-      ? CLUSTER_DEFAULT_NETWORK_SETTINGS_IPV4.serviceNetworkCidr
-      : CLUSTER_DEFAULT_NETWORK_SETTINGS_IPV6.serviceNetworkCidr,
-    clusterNetworkHostPrefix: isIPv4
-      ? CLUSTER_DEFAULT_NETWORK_SETTINGS_IPV4.clusterNetworkHostPrefix
-      : CLUSTER_DEFAULT_NETWORK_SETTINGS_IPV6.clusterNetworkHostPrefix,
-  };
-};
-
-// TODO: Replace this by API call to retrieve the defaults from the backend
-export const getClusterDefaultSettings = (clusterNetworkCidr: string) => ({
-  ...getDefaultNetworkSettings(clusterNetworkCidr),
-});
-
-export const isAdvConf = (cluster: Cluster) => {
-  const defaultNetworkSettings = getDefaultNetworkSettings(cluster.clusterNetworkCidr as string);
-  return (
-    cluster.clusterNetworkCidr !== defaultNetworkSettings.clusterNetworkCidr ||
-    cluster.clusterNetworkHostPrefix !== defaultNetworkSettings.clusterNetworkHostPrefix ||
-    cluster.serviceNetworkCidr !== defaultNetworkSettings.serviceNetworkCidr
-  );
-};
+export const isAdvConf = (
+  cluster: Cluster,
+  defaultNetworkSettings: Partial<ClusterNetworkDefaultSettings>,
```

Comment:
Since the `ClusterNetworkDefaultSettings` is always used as Partial, maybe we can change the type to have the keys optional and stop using Partial?

Reply:
@jtomasek if `ClusterNetworkDefaultSettings` would contain 1000 properties I would end up making 1000 changes...
Using `Partial` looks more convenient IMHO.

Reply:
solved at 40e8458acf363613d6c773a5db5901ea5b129192